gradle-conjure/src/main/java/com/palantir/gradle/conjure/CompileIrTask.java

```diff
@@ -37,24 +38,40 @@
 @CacheableTask
 public class CompileIrTask extends DefaultTask {
     private static final String EXECUTABLE = OsUtils.appendDotBatIfWindows("bin/conjure");
+    private final RegularFileProperty outputIrFile = getProject().getObjects().fileProperty();
 
-    private File outputFile;
     private Supplier<File> inputDirectory;
     private Supplier<File> executableDir;
     private final SetProperty<ServiceDependency> productDependencies =
             getProject().getObjects().setProperty(ServiceDependency.class);
 
+    /**
+     * Eagerly set where to output the generated IR.
+     *
+     * @deprecated Use {@link #getOutputIrFile()} instead.
+     */
+    @Deprecated
     public final void setOutputFile(File outputFile) {
-        this.outputFile = outputFile;
+        outputIrFile.set(outputFile);
     }
 
-    public final void setInputDirectory(Supplier<File> inputDirectory) {
-        this.inputDirectory = inputDirectory;
+    /**
+     * Eagerly get where to output the generated IR.
+     *
+     * @deprecated Use {@link #getOutputIrFile()} instead.
+     */
+    @Deprecated
+    public final File getOutputFile() {
+        return outputIrFile.getAsFile().get();
     }
 
     @OutputFile
-    public final File getOutputFile() {
-        return outputFile;
+    public final RegularFileProperty getOutputIrFile() {
```

Comment:
can we put a `@PathSensitivity(NONE)` since the task doesn't care about the path?